### ISSUE TITLE
Logitech Receiver idProduct=0xaaad for MX Anywhere 2 Mouse

### ISSUE THREAD

**Issue by ksanger3838** [pwr-Solaar/Solaar#1046]
solaar -- version 1.0.4
solaar --show "No Devices Found"
My problem is I am running Fedora 32, with a Logitech MX Anywhwere Mouse that uses a USB Pico Receiver.   idVendor is 0x046d Logitech, Inc.   idProduct is 0xaaad.  
I don't believe that this receiver is currently supported.

I've been using this mouse for a long time. Its my favorite.  Recent updates to Fedora and the Logitech driver have broken the mouse.   The receiver is not found.   The mouse can not connect via USB Bluetooth Pico Receiver.  I've spent 3 hours trying to fix this.   linux is great until it isn't.  Nice update.  So I've tried solaar and ltunify without any luck.

Fedora version is 32.   Linux kernel is 5.9.16-100.fc32.x86_64

Output from lsusb:
```
Bus 001 Device 007: ID 046d:aaad Logitech, Inc. 
Couldn't open device, some information will be missing
Device Descriptor:
  bLength                18
  bDescriptorType         1
  bcdUSB               2.00
  bDeviceClass            0 
  bDeviceSubClass         0 
  bDeviceProtocol         0 
  bMaxPacketSize0        32
  idVendor           0x046d Logitech, Inc.
  idProduct          0xaaad 
  bcdDevice            3.01
  iManufacturer           1 
  iProduct                2 
  iSerial                 0 
  bNumConfigurations      1
  Configuration Descriptor:
    bLength                 9
    bDescriptorType         2
    wTotalLength       0x0022
    bNumInterfaces          1
    bConfigurationValue     1
    iConfiguration          4 
    bmAttributes         0x80
      (Bus Powered)
    MaxPower               98mA
    Interface Descriptor:
      bLength                 9
      bDescriptorType         4
      bInterfaceNumber        0
      bAlternateSetting       0
      bNumEndpoints           1
      bInterfaceClass         3 Human Interface Device
      bInterfaceSubClass      0 
      bInterfaceProtocol      0 
      iInterface              0 
        HID Device Descriptor:
          bLength                 9
          bDescriptorType        33
          bcdHID               1.11
          bCountryCode            0 Not supported
          bNumDescriptors         1
          bDescriptorType        34 Report
          wDescriptorLength      25
         Report Descriptors: 
           ** UNAVAILABLE **
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x81  EP 1 IN
        bmAttributes            3
          Transfer Type            Interrupt
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0020  1x 32 bytes
        bInterval               1
```

dmesg returns:
```
[   76.574487] hid-generic 0003:046D:AAAD.0006: hiddev96,hidraw0: USB HID v1.11 Device [Logitech USB BootLoader] on usb-0000:00:12.2-2.1/input0
[   84.776946] usb 1-2.1: USB disconnect, device number 3
[   90.582207] usb 1-2.1: new full-speed USB device number 6 using ehci-pci
[   90.664536] usb 1-2.1: New USB device found, idVendor=046d, idProduct=aaad, bcdDevice= 3.01
[   90.664541] usb 1-2.1: New USB device strings: Mfr=1, Product=2, SerialNumber=0
[   90.664544] usb 1-2.1: Product: USB BootLoader
[   90.664546] usb 1-2.1: Manufacturer: Logitech
[   90.667835] hid-generic 0003:046D:AAAD.0007: hiddev96,hidraw0: USB HID v1.11 Device [Logitech USB BootLoader] on usb-0000:00:12.2-2.1/input0
[   90.720423] hid-generic 0003:046D:AAAD.0008: hiddev96,hidraw0: USB HID v1.11 Device [Logitech USB BootLoader] on usb-0000:00:12.2-2.1/input0
[  106.790010] usb 1-2.1: USB disconnect, device number 6
[  109.011214] usb 1-2.1: new full-speed USB device number 7 using ehci-pci
[  109.092579] usb 1-2.1: New USB device found, idVendor=046d, idProduct=aaad, bcdDevice= 3.01
[  109.092584] usb 1-2.1: New USB device strings: Mfr=1, Product=2, SerialNumber=0
[  109.092587] usb 1-2.1: Product: USB BootLoader
[  109.092589] usb 1-2.1: Manufacturer: Logitech
[  109.097891] hid-generic 0003:046D:AAAD.0009: hiddev96,hidraw0: USB HID v1.11 Device [Logitech USB BootLoader] on usb-0000:00:12.2-2.1/input0
[  109.161281] hid-generic 0003:046D:AAAD.000A: hiddev96,hidraw0: USB HID v1.11 Device [Logitech USB BootLoader] on usb-0000:00:12.2-2.1/input0
[  164.834760] nf_conntrack: default automatic helper assignment has been turned off for security reasons and CT-based  firewall rule not found. Use the iptables CT target to attach helpers instead.
[  375.468577] usb 2-4: 3:1: cannot get min/max values for control 2 (id 3)
```

**Comment by pfps** [comment]
It looks as if your receiver is in firmware update mode, but I know little about updating the firmware on Logitech Unifying receivers.  Indications that this is the case is the unusual Product Name (USB Bootloader) and Product ID (AAAD).  As far as I know, a Logitech Receiver that is in its normal mode should have a Product Name that has "Receiver" in it and a Product ID starting with C5.

@FFY00 does this sound right?

There are several related threads on firmware updates to Logitech Unifying Receivers, including https://github.com/fwupd/fwupd/issues/1018, which may indicate what is needed to get out of this state.

**Comment by FFY00** [comment]
Yes, this receiver is in bootload mode. You should be able to unplug it and plug it again to go back to normal mode. If it doesn't go back, it means something went wrong and the receiver doesn't have valid firmware, in which case you can use https://github.com/fwupd/fwupd to get it working normally again.

**Comment by pfps** [comment]
@ksanger3838 What is the current situation?

**Comment by ksanger3838** [comment]
Yes we were able to upgrade the Logitech Dongle Firmware and load solar to get the MX Anywhere Mouse and Logitech Unifying Receiver running.   Though we have moved on to using a usb bluetooth adapter.

Thank you guys for all your help.   I believe that the Unifying Receiver was left in an update state after starting a software firmware update from the gnome notification panel.   I do not recall the update asking for super user password for the firmware update.  We tried this first after another software update must have caused the Logitech Unifying Receiver to fail.

To fix this we unplugged and reinstalled the Logitech USB Dongle.   
Then to update the Logitech Dongle firmware we did: 
`sudo fwupdmgr get-updates
sudo fwupdmgr update`
We were hesitant to use fwupdmgr as there was no command to force it to only work on the Logitech Unifying Receiver.   `sudo fwupdmgr get-devices` showed two hard drives, an SSD drive, a Microsoft Dongle, and the Unifying Receiver (Logitech).  Thankfully only the receiver was updated.

After that we reinstalled solaar, reinstalled bluez, and used bluetoothctl to pair the Logitech MX Anywhere Mouse.
`sudo dnf install solaar
sudo dnf install bluez
bluetoothctl --help
bluetoothctl --devices
bluetoothctl pair XX:XX:XX:XX:XX:XX`    where XX... is the Logitech Unifying Receiver Device ID
`bluetoothctl connect XX:XX:XX:XX:XX:XX`

And we are up and running.   The mouse paired well, worked well, and repaired automatically when we powered it off and on, or rebooted.  This may have also fixed the mouse's "Gestures".   I believe that the built in gestures might be the source of an aggravation where the mouse causes documents, text pages, and spreadsheets to move to the bottom right of the page or sheet.   Over the brief time we were using solaar this stopped happening.

Note at one point before installing solaar we had also performed `sudo modprobe hid-logitech-dj` which may be needed.  

While we were working on the Unifying Receiver we had purchased an ASUS USB-BT500 bluetooth usb dongle.  Installing this dongle took four days to get running.  We now know more about linux modules than we want.  Currently with the ASUS bluetooth usb dongle `lsmod | grep solaar` and `lsmod | grep logitech` do not show solaar nor hid-logitech-dj as being loaded.  In addition to the mouse the usb bluetooth module lets us pair a headset.  Hopefully we're good until another SW change forces us to find another fix

Thanks again.  Informing us that the dongle was stuck in a firmware update state helped us find a solution.
